deptrack-devops-examples/app/DevVM.hs

```diff
@@ -0,0 +1,134 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TupleSections     #-}
+module Main where
+
+import           Control.Monad             (forM, mapM, void)
+import           Data.Monoid
+import           Data.String.Conversions   (convertString)
+import           Data.Text                 (pack)
+import           DepTrack
+import           Devops.Base               (DevOp, noop)
+import           Devops.Bootstrap.Build
+import           Devops.Bootstrap.DO
+import           Devops.Bootstrap.Parasite
+import           Devops.Callback           (binaryCall, continueConst)
+import           Devops.Cli                (App (..), Method (..), SelfPath,
+                                            appMain, appMethod, methodArg,
+                                            simpleMain)
+import           Devops.Debian.Base        (deb)
+import           Devops.Debian.Commands    (git)
+import           Devops.Debian.User        (Group (..), User (..),
+                                            directoryPermissions, homeDirPath,
+                                            mereUser, preExistingUser,
+                                            userDirectory)
+import           Devops.Git                (GitRepo (..), gitClone, gitDir)
+import           Devops.Haskell            (stackPackage)
+import           Devops.Optimize           (optimizeDebianPackages)
+import           Devops.OS
+import           Devops.Ref
+import           Devops.Storage            (DirectoryPresent (..), FileContent,
+                                            FileLinked, FilePresent (..),
+                                            directory, fileContent, fileLink,
+                                            preExistingFile, subdirectory,
+                                            (</>))
+import           Network.DO                hiding (error)
+import           System.Environment        (getArgs)
+import           System.IO                 (BufferMode (..), hSetBuffering,
+                                            stderr, stdout)
+
+
+-- | Stages of execution of this application
+data Stage = Local String Int
+           -- ^ The binary was called (e.g., by a human) on the local-host.
+           | Remote
+           -- ^ The binary was called remotely
+
+parseStage :: [String] -> (Stage, Method)
+parseStage = \case
+    ("_remote_":arg:[]) -> (Remote, appMethod arg)
+    (host:key:arg:[])   -> (Local host (read key),  appMethod arg)
+    args                -> error $ "unparsed args: " ++ show args
+
+unparseStage :: Stage -> Method -> [String]
+unparseStage stage m = case stage of
+    Local h k -> [ h, show k, methodArg m ]
+    Remote    -> [ "_remote_", methodArg m ]
+
+--------------------------------------------------------------------
+
+main :: IO ()
+main = do
+  hSetBuffering stdout LineBuffering
+  hSetBuffering stderr LineBuffering
+  let app = App parseStage unparseStage stages [optimizeDebianPackages] :: App Stage
+  appMain app
```

Comment:
4-chars indent as well (or keep 2 everywhere in this file for consistency -- I mind less in example files)